src/webgpu/api/validation/capability_checks/limits/maxBufferSize.spec.ts

```diff
@@ -0,0 +1,28 @@
+import { kLimitBaseParams, makeLimitTestGroup } from './limit_utils.js';
+
+const limit = 'maxBufferSize';
+export const { g, description } = makeLimitTestGroup(limit);
+
+g.test('createBuffer,at_over')
+  .desc(`Test using at and over ${limit} limit`)
+  .params(kLimitBaseParams)
+  .fn(async t => {
+    const { limitTest, testValueName } = t.params;
+    await t.testDeviceWithRequestedLimits(
+      limitTest,
+      testValueName,
+      async ({ device, testValue, actualLimit, shouldError }) => {
+        await t.testForValidationErrorWithPossibleOutOfMemoryError(
+          () => {
+            const buffer = device.createBuffer({
+              usage: GPUBufferUsage.VERTEX,
+              size: testValue,
+            });
+            buffer.destroy();
```

Comment:
Usually we avoid having destroy() calls inside the same error scope with the call we're testing, but in this case it's annoying enough to fix that I don't think it's worth fixing.

Reply:
Also I don't know why I didn't think about this until now.